api/v3/Attachment.php

```diff
@@ -147,6 +147,7 @@ function civicrm_api3_attachment_create($params) {
   $fileDao->copyValues($file);
   if (!$id) {
     $fileDao->uri = CRM_Utils_File::makeFileName($name);
+    $fileDao->created_id = CRM_Core_Session::getLoggedInContactID();
```

Comment:
I think it's better to do this on the spec function - ie 
    'api.default' => 'user_contact_id',

Reply:
yeah right .. how could I overlooked it :( Will do it